### DISCUSSION TITLE
Midi Stopped Working After Software Upgrade to Version 3.4.2-1

### DISCUSSION THREAD

**Discussion by OrgelSpieler** [GrandOrgue/grandorgue#866] [Help] [answered]
I've been successfully running GrandOrgue verions 0.3.1.2330-156.3 for sometime on a Win 7 Platform with RME sound cards.  I decided to get my instrument on the latest version 3.4.2-1.  After installing, GrandOrgue no longer responds to my midi instrument.  The midi monitor shows the commands are being received so I'm not sure what broke.  Activating a key or stop on the console screen results in the correct response on the display and plays audio where applicable.   If I downgrade to my older version, everything works properly.  What did I overlook with this newer version?  Thank you.

Paul

**Comment by Jean-Gabriel1982** [comment]
Hi, 
What do you have in File>Settings>Audio output ? I had this problem before (now another one) and Audio output was wrong

**Reply by OrgelSpieler** [reply]
I checked this out and all the settings are correct.  The audio is functioning as I am able to playback a midi recording or press the keys on the screen and get an output.  Thanks for taking time to respond to my inquiry.

**Comment by oleg68** [comment, marked as answer]
Seems you have to adjust your midi settings.
1. Open File->Settings->Midi and make sure that all your midi devices are present and are selected
2. If it still doesn't work, make midi assignments for all manuals and the pedal.

**Reply by OrgelSpieler** [reply]
Thanks for taking time to answer my question.

My devices are present and selected.  I redid the midi assignment for a manual and that resulted in sound output, but it revealed a larger issue.  It seems the upgrade changed the names of my midi devices.  Under my old version (3.1.2330-156), my midi input device was shown as AIO Midi 0.  When I did the upgrade (3.4.2-1), my midi device shows as winnm:AIO Midi 0 (note the prefix "winnm" has been added).   Since all of my midi settings were set up for AIO Midi 0, the system could not map the inputs because it is expecting winnm:AIO Midi 0.   If all I had to do was redo the assignments for the manuals and pedal, it would be no big deal.  But I have assignments for 156 physical stops, couplers and pistons (both send and receive).  To redo all of that is quite an undertaking.

This begs these questions:
1.  Does anyone know why the naming convention convention changed for my midi devices  (RME AIO Card) on the upgrade? 
2.  Is there anyway to force the system to use my older midi names?
3.  If the new names must be used, is there a file that can be edited so the change can be done efficiently?

**Reply by OrgelSpieler** [reply]
I'm going to mark this as answered and start a new discussion.
